app/controllers/early_results_controller.rb

```diff
@@ -0,0 +1,29 @@
+class EarlyResultsController < ApplicationController
```

Comment:
This feels like it is outside the normal flow, which is really hard to reason about. I think we have a decent data model in here - maybe it should be made more explicit in a refactoring prior to this work? If the data model were more obvious in the code, maybe this work could be either incorporated into it or use an 'adapter' pattern to extend it if that it what we need to do? [In CFE civil, we moved the calculation into one place, and then wrote an adapter to take the 'wrong' result and correct it based on the (possible) incompleteness of the data)

Reply:
Originally when we were speccing this out with design, we were thinking of implementing a screen that could pop up at any point in the flow and signal an early ineligible result. Because of that, and because of the way we currently generate the different sections of steps, we went with an idea of leaving the flow, and hooking back into it again if the user decided to resume their check and add more information. For a V1 implementation we went with showing the interruption screen at the end of certain sections (with the understanding that we'd like to throw this interruption screen into the flow at potentially any point in the future) but maybe leaving the flow makes less sense in this current V1 scope as we know where the screen will be. I'm not sure about the adapter pattern but would be interested to hear more :)

Reply:
Putting it in at any point does mess up the existing (software) design somewhat. Wonder if there would be any mileage in having 3 screens with the same content to simplify things? Could the step be dependant on the CFE result somehow? [If its only at the end of the sections, then think a simple version of adding a screen works]

Reply:
We did actually have a go at spiking something like that out in one of our earlier attempts! We added the relevant screen section and the idea was to have a different form/view for each screen type - something like this commit: https://github.com/ministryofjustice/laa-check-client-qualifies/pull/1089/commits/a10db289d654c6e6cad1764b3ebf52047bffda2f

We moved away from that idea when it seemed we might be showing the screen at any point in the journey, and also weren't sure if we should duplicate the screen logic for the different screen types (the continuous check, where the screen could pop up at any point is still the aim after the V1 implementation as far as I'm aware, but that might change). I'm not against adding in a screen that follows the normal pattern though if we think it is the better way to go. Just might be harder to refactor later on I guess.